proof_of_concept/components/asset_store.py

```diff
@@ -33,10 +49,17 @@ def store(self, asset: Asset) -> None:
         if asset.id in self._assets:
             raise KeyError(f'There is already an asset with id {id}')
 
-        self._assets[asset.id] = asset
+        self._assets[asset.id] = copy(asset)
```

Comment:
looking back and forth in the library (because i don't know it), this seems ok

Reply:
The idea here is that because the asset image is moved or copied, I have to update the Asset object to point to the new location. I don't want to mess up the Asset object passed by the caller, so I'm copying it before modifying. Previously that wasn't necessary, because the Asset wasn't altered.